habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -443,23 +448,42 @@ def _load_navmesh(self, ep_info):
         base_dir = osp.join(*ep_info.scene_id.split("/")[:2])
 
         navmesh_path = osp.join(base_dir, "navmeshes", scene_name + ".navmesh")
-        self.pathfinder.load_nav_mesh(navmesh_path)
 
-        self._navmesh_vertices = np.stack(
-            self.pathfinder.build_navmesh_vertices(), axis=0
-        )
-        self._island_sizes = [
-            self.pathfinder.island_radius(p) for p in self._navmesh_vertices
-        ]
-        self._max_island_size = max(self._island_sizes)
-        self._largest_island_idx = self.pathfinder.get_island(
-            self._navmesh_vertices[np.argmax(self._island_sizes)]
+        if osp.exists(navmesh_path):
+            self.pathfinder.load_nav_mesh(navmesh_path)
+            logger.info(f"Loaded navmesh from {navmesh_path}")
+        else:
+            logger.warning(
+                f"Requested navmesh to load from {navmesh_path} does not exist. Recomputing from configured values and caching."
+            )
+            navmesh_settings = NavMeshSettings()
+            navmesh_settings.set_defaults()
+
+            agent_config = None
+            if hasattr(self.habitat_config.agents, "agent_0"):
+                agent_config = self.habitat_config.agents.agent_0
+            elif hasattr(self.habitat_config.agents, "main_agent"):
+                agent_config = self.habitat_config.agents.main_agent
+            else:
+                raise ValueError(f"Cannot find agent parameters.")
+            navmesh_settings.agent_radius = agent_config.radius
+            navmesh_settings.agent_height = agent_config.height
+            navmesh_settings.agent_max_climb = agent_config.max_climb
+            navmesh_settings.agent_max_slope = agent_config.max_slope
+            navmesh_settings.include_static_objects = True
+            self.recompute_navmesh(self.pathfinder, navmesh_settings)
+            os.makedirs(osp.dirname(navmesh_path), exist_ok=True)
+            self.pathfinder.save_nav_mesh(navmesh_path)
+
+        # NOTE: allowing indoor islands only
+        self._largest_island_idx = get_largest_island_index(
+            self.pathfinder, self, allow_outdoor=False
         )
 
     @property
     def largest_island_idx(self) -> int:
```

Comment:
If it is not too much work, I would rename `largest_island_idx` to `largest_indoors_island_idx`.